### DISCUSSION TITLE
Removing generated files

### DISCUSSION THREAD

**Discussion by MillerAdulu** [droidconKE/flutterconKEApp#18] [General]
I propose for us to consider **not** committing generated files:
- .freezed.dart
- .g.dart

The proposed reason for this is that as many people begin to work on the app and where there might be slight changes in the generated files, it usually becomes chaotic to merge. The easiest way to merge is usually to delete the conflicting files and regenerate them since it's not recommended to edit them by hand. 

For purposes of online builds, what can happen is that we can introduce a GitHub action to generate those files when the APK is being built. To add the gen command after the `Get dependencies` step. This way, we keep the codebase light and only files that require actual changing by hand. It can be argued that committing them prevents unnecessary build times on the CI/CD pipeline, but the effort to merge when a conflict arises is more trouble than building them on the CI/CD pipeline. 

For next steps, I recommend creating an issue that will open a PR to delete those files and in addition, update the docs to reflect this change and show how to go about the build. Many open-source projects have `Makefile`s which normally have already defined commands that can be run quickly or copied on the terminal and then referenced in the docs. It's not standard Flutter practice to have a script section like composer.json (PHP) and package.json (JS) so a Makefile is the best possible solution. 

Here's a small sample from a Makefile I use for a couple of projects. 

```Makefile
gen:
		dart run build_runner build --delete-conflicting-outputs

fmt:
		dart fix --apply && dart format lib test

localize:
		flutter gen-l10n

apk:
		flutter build apk  --flavor production --target lib/main_production.dart -vv
		
pods:
		cd ios && pod install --repo-update --verbose && cd ..
```

**Comment by MillerAdulu** [comment]
Resolved in https://github.com/droidconKE/flutterconKEApp/pull/19

**Reply by mboyamike** [reply]
Great!
